ckanapi/remoteckan.py

```diff
@@ -82,12 +91,24 @@ def call_action(self, action, data_dict=None, context=None, apikey=None,
         if self.get_only:
             status, response = self._request_fn_get(url, data_dict, headers, requests_kwargs)
         else:
-            status, response = self._request_fn(url, data, headers, files, requests_kwargs)
+            status, response = self._request_fn(url, data, headers, files,
+                                                progress, requests_kwargs)
         return reverse_apicontroller_action(url, status, response)
 
-    def _request_fn(self, url, data, headers, files, requests_kwargs):
-        r = self.session.post(url, data=data, headers=headers, files=files,
-            allow_redirects=False, **requests_kwargs)
+    def _request_fn(self, url, data, headers, files, progress, requests_kwargs):
+        if files:  # use streaming
+            files['upload'] = (getattr(files['upload'], 'name', 'upload'),
+                               files['upload'])
```

Comment:
I am not completely clear what thsi field name could be; it appears it can be anything and it doesn't matter what it is. I now also cover the case that the the key passed for files is already a key in data_dict, which would potentially silently screw up the upload.